File: 867. Transpose Matrix/solution.js
/**
 * @param {number[][]} matrix
 * @return {number[][]}
 */
 var transpose = function(matrix) {
  let result = [];
  for (let column = 0; column < matrix[0].length; column++){
    let newRow = [];
    for (let row = 0; row < matrix.length; row++){
      newRow.push(matrix[row][column]);
    }
    result.push(newRow);
  }
  return result;
};

File: 867. Transpose Matrix/solution.cpp
class Solution {
public:
  vector<vector<int>> transpose(vector<vector<int>>& matrix) {
    vector<vector<int>> result;
    for (int column = 0; column < matrix[0].size(); column++){
      vector<int> newRow;
      for (int row = 0; row < matrix.size(); row++){
        newRow.push_back(matrix[row][column]);
      }
      result.push_back(newRow);
    }
    return result;
  }
};